internal/job/integration/redaction_integration_test.go

```diff
@@ -0,0 +1,137 @@
+		c.Exit(0)
+	})
+
+	err = tester.Run(t)
+	if err != nil {
+		t.Fatalf("running executor tester: %v", err)
+	}
+
+	if !strings.Contains(tester.Output, "The agent token is: [REDACTED]") {
+		t.Fatalf("expected agent token to be redacted, but it wasn't. Full output: %s", tester.Output)
+	}
+}
+
+func TestRedactorAdd_RedactsVarsAfterUse(t *testing.T) {
+	// when this test is run inside an agent (ie in CI), we don't want to connect to the job API of the executor running this test
+	// we want to instead connect to the job API of the executor process we're testing
+	t.Setenv("BUILDKITE_AGENT_JOB_API_TOKEN", "")
+	t.Setenv("BUILDKITE_AGENT_JOB_API_SOCKET", "")
+
+	tester, err := NewExecutorTester(mainCtx)
+	if err != nil {
+		t.Fatalf("setting up executor tester: %v", err)
+	}
+	defer tester.Close()
+
+	secret := "huunter2" // secrets 6 chars or shorter don't get redacted, hence the extra u
+	tester.ExpectGlobalHook("command").AndCallFunc(redactionTestCommandHook(t, secret))
+
+	err = tester.Run(t)
+	if err != nil {
+		t.Fatalf("running executor tester: %v", err)
+	}
+
+	// verify that the secret was echoed prior to calling RedactionCreate
+	if !strings.Contains(tester.Output, fmt.Sprintf("The secret is: %s", secret)) {
+		t.Fatalf("expected secret to be echoed prior to redaction, but it wasn't. Full output: %s", tester.Output)
+	}
+
+	// now check that the string got redacted after the call to RedactionCreate
+	if !strings.Contains(tester.Output, "There should be a redacted here: [REDACTED]") {
+		t.Fatalf("expected secret to be redacted after RedactionCreate, but it wasn't. Full output: %s", tester.Output)
+	}
+}
+
+func TestRedactorAdd_FailsWhenNoInitialRedactionStringsArePresent(t *testing.T) {
+	// when this test is run inside an agent (ie in CI), we don't want to connect to the job API of the executor running this test
+	// we want to instead connect to the job API of the executor process we're testing
+	t.Setenv("BUILDKITE_AGENT_JOB_API_SOCKET", "")
+	t.Setenv("BUILDKITE_AGENT_JOB_API_TOKEN", "")
+
+	tester, err := NewExecutorTester(mainCtx)
+	if err != nil {
+		t.Fatalf("setting up executor tester: %v", err)
+	}
+	defer tester.Close()
+
+	secret := "huunter2" // secrets 6 chars or shorter don't get redacted, hence the extra u
+	tester.ExpectGlobalHook("command").AndCallFunc(redactionTestCommandHook(t, secret))
+
+	err = tester.Run(t, `BUILDKITE_REDACTED_VARS=""`)
+	if err != nil {
+		t.Fatalf("running executor tester: %v", err)
+	}
+
+	// verify that the secret was echoed prior to calling RedactionCreate
+	if !strings.Contains(tester.Output, fmt.Sprintf("The secret is: %s", secret)) {
+		t.Fatalf("expected secret to be echoed prior to redaction, but it wasn't. Full output: %s", tester.Output)
+	}
+
+	// now check that the string is still unredacted after the call to RedactionCreate
+	if !strings.Contains(tester.Output, fmt.Sprintf("There should be a redacted here: %s", secret)) {
```

Comment:
i wonder if it's clearer (if less idiomatic) to just have this be
```suggestion
	if !strings.Contains(tester.Output, "There should be a redacted here: huunter2" {

```

Reply:
It's not less idiomatic, if anything I think it's more idiomatic for a test on `strings.Contains` to have the full string being searched for plainly (e.g. you can grep the repo for the whole string to find the test).